daffodil-japi/src/test/java/edu/illinois/ncsa/daffodil/example/TestJavaAPI.java

```diff
@@ -58,803 +58,806 @@
+        ProcessorFactory pf = c.compileFile(schemaFile);
+        DataProcessor dp = pf.onPath("/");
+        dp.setDebugger(debugger);
+        dp.setDebugging(true);
+
+        // Serialize the parser to memory, then deserialize for parsing.
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        WritableByteChannel output = Channels.newChannel(os);
+        dp.save(output);
+
+        ByteArrayInputStream is = new ByteArrayInputStream(os.toByteArray());
+        ReadableByteChannel input = Channels.newChannel(is);
+        edu.illinois.ncsa.daffodil.japi.Compiler compiler = Daffodil.compiler();
+        DataProcessor parser = compiler.reload(input);
+
+        try {
+            parser.setValidationMode(ValidationMode.Full);
+            fail();
+        } catch (InvalidUsageException e) {
+            assertEquals("'Full' validation not allowed when using a restored parser.", e.getMessage());
+        }
+
+        // reset the global logging and debugger state
+        Daffodil.setLogWriter(new ConsoleLogWriter());
+        Daffodil.setLoggingLevel(LogLevel.Info);
+    }
+
+    @Test
+    public void testJavaAPI2() throws IOException {
+        LogWriterForJAPITest lw = new LogWriterForJAPITest();
+
+        Daffodil.setLogWriter(lw);
+        Daffodil.setLoggingLevel(LogLevel.Info);
+
+        edu.illinois.ncsa.daffodil.japi.Compiler c = Daffodil.compiler();
+        c.setValidateDFDLSchemas(false);
+        java.io.File schemaFile = getResource("/test/japi/mySchema1.dfdl.xsd");
+        ProcessorFactory pf = c.compileFile(schemaFile);
+        DataProcessor dp = pf.onPath("/");
+        java.io.File file = getResource("/test/japi/myDataBroken.dat");
+        java.io.FileInputStream fis = new java.io.FileInputStream(file);
+        java.nio.channels.ReadableByteChannel rbc = java.nio.channels.Channels.newChannel(fis);
+        JDOMInfosetOutputter outputter = new JDOMInfosetOutputter();
+        ParseResult res = dp.parse(rbc, outputter);
+
+        // TODO: NEED a java friendly way to get the status of the outputter.
+        // Scala enums don't work well
+        // assertTrue(outputter.getStatus() != Status.DONE); // This is a hack
+        // so that Java doesn't have to know about Nope/Maybe, need to figure
+        // out better api that is Java compatible
+        assertTrue(res.isError());
+        java.util.List<Diagnostic> diags = res.getDiagnostics();
+        assertEquals(1, diags.size());
+        Diagnostic d = diags.get(0);
+        // System.err.println(d.getMessage());
+        assertTrue(d.getMessage().contains("int"));
+        assertTrue(d.getMessage().contains("Not an int"));
+        assertTrue(d.getDataLocations().toString().contains("10"));
+        java.util.List<LocationInSchemaFile> locs = d.getLocationsInSchemaFiles();
+        assertEquals(1, locs.size());
+        LocationInSchemaFile loc = locs.get(0);
+        assertTrue(loc.toString().contains("mySchema1.dfdl.xsd"));
```

Comment:
This is the actual change.  Was "mySchema2.dfdl.xsd" which is arguably better.

A side effect of the change to resolve everything from Term, is that we're always pointing at a term when we issue an SDE, and never pointing at a global element decl, or type definition. This is not correct. Somehow we're not getting the right object to issue the SDE. When we resolve properties, the Found object also includes the location where that property was found - seems we aren't utilizing that information to get a more specific object to use to issue the SDE.